handlers_racing_test.go

```diff
@@ -182,20 +183,37 @@ func TestSignaturePass(t *testing.T) {
 				case "/sign/file":
 					// use the margo pkg to calculate the signable block of the mar file
 					// as input for the signature verification
-					rawInput, _ := base64.StdEncoding.DecodeString(testcase.signaturerequests[j].Input)
+					rawInput, err := base64.StdEncoding.DecodeString(testcase.signaturerequests[j].Input)
+					if err != nil {
+						t.Fatalf("in test case %d on endpoint %q, error '%v' in response %d;\nrequest was: %+v\nresponse was: %+v failed to decode input",
+							i, testcase.endpoint, err, j, testcase.signaturerequests[j], response)
+					}
+					if !bytes.Equal(miniMarB, rawInput) {
+						t.Fatalf("in test case %d on endpoint %q, error '%v' in response %d;\nrequest was: %+v\nresponse was: %+v decoded input did not match expected input",
+							i, testcase.endpoint, err, j, testcase.signaturerequests[j], response)
+					}
 					var marFile margo.File
-					margo.Unmarshal(rawInput, &marFile)
+					err = margo.Unmarshal(rawInput, &marFile)
+					if err != nil {
+						t.Fatalf("in test case %d on endpoint %q, error '%v' in response %d;\nrequest was: %+v\nresponse was: %+v failed to unmarshal mar sig",
+							i, testcase.endpoint, err, j, testcase.signaturerequests[j], response)
+					}
 					rawKey, err := base64.StdEncoding.DecodeString(response.PublicKey)
 					if err != nil {
 						t.Fatalf("in test case %d on endpoint %q, error '%v' in response %d;\nrequest was: %+v\nresponse was: %+v",
 							i, testcase.endpoint, err, j, testcase.signaturerequests[j], response)
 					}
-					key, err := x509.ParsePKIXPublicKey(rawKey)
+					pubkey, err := x509.ParsePKIXPublicKey(rawKey)
 					if err != nil {
 						t.Fatalf("in test case %d on endpoint %q, error '%v' in response %d;\nrequest was: %+v\nresponse was: %+v",
 							i, testcase.endpoint, err, j, testcase.signaturerequests[j], response)
 					}
-					err = marFile.VerifySignature(key)
+					err = marFile.VerifySignature(pubkey)
+					// TODO: figure out why testcase 6 fails with "error verifying mar signature 'no valid signature found'"
+					if err == nil {
```

Comment:
Failed on its own in https://github.com/mozilla-services/autograph/pull/763 so it's not related to other changes.